shared/teams/team/settings-tab/channel-popup.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react'
+import * as Kb from '../../../common-adapters'
+import * as Styles from '../../../styles'
+import * as Constants from '../../../constants/teams'
+import * as Types from '../../../constants/types/teams'
+import * as Container from '../../../util/container'
+import {pluralize} from '../../../util/string'
+
+type Props = {
+  disabledChannels?: Array<string>
+  onCancel: () => void
+  onComplete: (channels: Array<string>) => void
+  teamID: Types.TeamID
+}
+
+const ChannelPopup = (props: Props) => {
+  const {disabledChannels, onCancel, onComplete, teamID} = props
+  const [filter, setFilter] = React.useState('')
+  const filterLCase = filter.toLowerCase()
+  const onChangeFilter = (value: string) => setFilter(value)
+
+  const channelInfos = Container.useSelector(s => Constants.getTeamChannelInfos(s, teamID))
+  const channels = [...channelInfos.values()].map(ci => ci.channelname)
+  const channelsFiltered = filter ? channels.filter(c => c.toLowerCase().includes(filterLCase)) : channels
+
+  const [selected, setSelected] = React.useState(new Set<string>())
+  const onSelect = (channelname: string) => {
+    if (selected.has(channelname)) {
+      selected.delete(channelname)
+    } else {
+      selected.add(channelname)
+    }
+    setSelected(new Set(selected))
```

Comment:
why do we need the `new Set()` here?

Reply:
Otherwise the reference stays the same and react doesn't trigger an update.